createtopics_test.go

```diff
@@ -0,0 +1,241 @@
+package kafka
```

Comment:
Is this file expected to be here? It seems weird to have those structures in a test file, and it’s unclear how it’s related to the kafka.Message changes.

Reply:
yes.  The existing test cases read and publishes messages to partition 0.  Consequently, there seemed to be no way to test whether or not partition was being set correctly or if it was a byproduct of partition 0 sharing go's zero value.

My solution was to implement createTopics in the test package only so I could ensure I was working on a topic with more than 1 partition.

Reply:
And to clarify, I left them in test because no production code had need for createTopics.  I figured they can always be pulled in if the need arises.